app/controllers/users_controller.rb

```diff
@@ -12,16 +12,30 @@ def new
   end
 
   def create
-    user = User.create(user_params)
-    if user.save
-      session[:user_id] = user.id
+    @user = User.create(user_params)
+    if @user.save
+      RegistrationMailer.with(user: @user).confirmation_email.deliver_now
+      session[:user_id] = @user.id
+      flash[:notice] = "Logged in as #{@user.first_name} #{@user.last_name}. This account has not yet been activated. Please check your email."
       redirect_to dashboard_path
     else
       flash[:error] = 'Username already exists'
       render :new
     end
   end
 
+  def update
+    @user = User.find(params[:id])
+    session[:user_id] = @user.id
+    @user.update(active: true)
+    flash[:notice] = "Thank you! Your account is now activated. Status: Active"
+    redirect_to dashboard_path
+  end
+
+  def activate
+    update
+  end
```

Comment:
so this is because a `post request` is needed to hit `update` from routes?

Reply:
That is what it seems. Maybe a good question for Ian tomorrow: "Which is worse? Using a get request with a non restful controller action or using an update action with a get request"